administrative-sdk/connection.js

```diff
@@ -233,43 +227,32 @@ class Connection {
    * @param {string} [auth] The authorization header value to pass along with the request.
    */
   _ajaxPost(url, formdata, auth) {
-    var request = new XMLHttpRequest({mozSystem: true});
-    var response = null;
-    request.open('POST', url);
+    var headers = new Headers();
     if (typeof auth !== 'undefined') {
-      request.setRequestHeader('Authorization', auth);
+      headers.append('Authorization', auth);
     }
     if (typeof formdata === 'string') {
-      // Send JSON by default
-      request.setRequestHeader('Content-Type',
+      headers.append('Content-Type',
         'application/json; charset=utf-8');
     }
-    return new Promise(function(resolve, reject) {
-      request.onload = function() {
-        response = Connection._parseResponse(request.responseText);
-        if (request.status >= 100 && request.status < 300) {
-          resolve(response);
-        } else {
-          reject(
-            {
-              status: request.status,
-              errors: response
+    var options = {
+      method: 'POST',
+      headers: headers,
+      body: formdata
+    };
+    return fetch(url, options)
+      .then(response =>
+        (response.json()
+          .then(data => {
+            if (response.ok) {
+              return data;
             }
-          );
-        }
-      };
-
-      request.onerror = function() {
-        reject(
-          {
-            status: request.status,
-            errors: response
-          }
-        );
-      };
-
-      request.send(formdata);
-    });
+            throw data;
+          })
+      ))
+      .catch(error => {
+        throw error;
+      });
```

Comment:
Same as in `_ajaxGet`. Although I do think some code could be shared between those two.